wagon.py

```diff
@@ -0,0 +1,1014 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import os
+import re
+import sys
+import time
+import json
+import shutil
+import urllib
+import urllib2
+import tarfile
+import zipfile
+import logging
+import platform
+import tempfile
+import subprocess
+import pkg_resources
+from threading import Thread
+from contextlib import closing
+
+import click
+try:
+    import virtualenv
+except ImportError:
+    pass
+from wheel import pep425tags as wheel_tags
+
+
+REQUIREMENT_FILE_NAMES = ['dev-requirements.txt', 'requirements.txt']
+METADATA_FILE_NAME = 'package.json'
+DEFAULT_WHEELS_PATH = 'wheels'
+
+DEFAULT_INDEX_SOURCE_URL = 'https://pypi.python.org/pypi/{0}/json'
+IS_VIRTUALENV = hasattr(sys, 'real_prefix')
+
+PLATFORM = sys.platform
+IS_WIN = (os.name == 'nt')
+IS_DARWIN = (PLATFORM == 'darwin')
+IS_LINUX = PLATFORM.startswith('linux')
+
+PROCESS_POLLING_INTERVAL = 0.1
+
+
+def setup_logger():
+    handler = logging.StreamHandler(sys.stdout)
+    formatter = logging.Formatter(
+        '%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger = logging.getLogger('wagon')
+    logger.addHandler(handler)
+    logger.setLevel(logging.INFO)
+    return logger
+
+
+logger = setup_logger()
+
+
+class PipeReader(Thread):
+    def __init__(self, fd, proc, logger, log_level):
+        Thread.__init__(self)
+        self.fd = fd
+        self.proc = proc
+        self.logger = logger
+        self.log_level = log_level
+        self.aggr = ''
+
+    def run(self):
+        while self.proc.poll() is None:
+            output = self.fd.readline()
+            if len(output) > 0:
+                self.aggr += output
```

Comment:
Already did in the python 3 branch
On Jul 6, 2016 07:03, "Nir Fuchs" <notifications@github.com> wrote:

> In wagon.py
> <https://github.com/cloudify-cosmo/wagon/pull/46#discussion_r69672687>:
>
> > +
> > +
> > +class PipeReader(Thread):
> > +    def __init__(self, fd, proc, logger, log_level):
> > +        Thread.__init__(self)
> > +        self.fd = fd
> > +        self.proc = proc
> > +        self.logger = logger
> > +        self.log_level = log_level
> > +        self.aggr = ''
> > +
> > +    def run(self):
> > +        while self.proc.poll() is None:
> > +            output = self.fd.readline()
> > +            if len(output) > 0:
> > +                self.aggr += output
>
> adding to Dan's note, treat all string as Unicode strings for Python 2 and
> 3 compatibility. (see from *future* import unicode_literals)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cloudify-cosmo/wagon/pull/46/files/1d672ac45d67d9f315af0d4e8dcb1feacb341305#r69672687>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB6QhKsxtzzko9kGULuv8X3yCyppuKbcks5qSykSgaJpZM4I0pgI>
> .
>